src/docs/autoscaling.md

```diff
@@ -0,0 +1,115 @@
+
+
+## Scaling instances ##
+A factor `desired_instances_count` is used to determine the desired number of instances to run for an App at an evaluated period. This factor is calculated differently for our three types of apps and is bounded by:
+ 
+`min_instance_count<=desired_instances_count<=max_instance_count`
+
+The definition of `min_instance_count` and `max_instance_count` can be found in the latter section. 
+
+### Delivery workers ###
+The autoscaler checks the total number of messages in the queues (ApproximateNumberOfMessages) of the app and scaled accordingly. One app can use multiple queues. This scaling is applicable to database tasks, CSV job processor, notifications senders and various research and priority jobs. 
+
+For example, 
+```notify-delivery-worker-sender``` is an app for delivering the notifications to the service providers, i.e. AWS SES and text messaging providers. The celery worker consumes job from two queues: ```send-sms-tasks``` and ```send-sms-tasks``` are being monitored. The number of instances is then calculated as:-
+
+```
+desired_instances_count = total_message_count / request_per_instance
+```
+where `total_message_count` is the sum of all the queues of this app and `request_per_instance` is a predefined estimated number of jobs an app instance can handle at one time.
+
+### Request handlers ##
+
+The inbound apps are scaled based on last 5 `request_counts`, where `request_counts` is the total number of incoming requests in one minute.  
+
+```
+desired_instance_count = max(request_counts)/request_per_instance
+```
+
+`request_per_instance` is a pre-defined scaler that indicates the number of requests that can be handled by one instance of app. 
+
+
+### Scheduled jobs ##
+Notify uses a proactive schedule for scheduled tasks. Scheduled tasks are CSV files the users uploaded to be run at a given time. We already have the information of job size (number of notificaitons) and the scheduled time. The proactive scheduler scales up the number of instances before the jobs start. This scaling applies to the same task as the Delivery workers above. 
+
+The desired number of instances for the particular app is calculated as follow:
+
+```
+desired_instance_count=num_of_scheduled_jobs/request_per_instance/scheduled_job_scaling_factor
+and
+min_instance_count<=desired_instances_count<=max_instance_count
+```
+
+The `num_of_scheduled_jobs` is the number of notifications to be sent in that task. `Request_per_instance` gives a first indication on the number of jobs that can be handled by one app instance. We can use `scheduled_job_scaling_factor ` to control how aggressively we want the scaling to be. Currently, this factor is set to 2 in Notify. Moreover, as the jobs are first delivered to the queues by delivery worker before these queues are consumed by senders and db workers. Hence, they are scaled up less agressively. 
```

Comment:
- The correct variable name is `scheduled_items` if you want to keep it consistent with the code.
- I don't think `request_per_instance` is relevant here
- `scheduled_job_scaling_factor` is not currently a variable in the app
- This factor is [currently set to 3](https://github.com/alphagov/notifications-paas-autoscaler/blob/master/main.py#L211)
- I know what you want to say in the last 2 sentences but I don't think it's very clear. "The reason we use a scaling factor is that not all the scheduled items are being put on the queue *at once*; Instead, each of them is processed and placed on the queue individually. That results in the queues not growing too fast and thus having need for a less aggressive scale."